### PR TITLE
assimp/5.1.x: Unbundle pugixml.

### PR DESCRIPTION
**Pull request by sh0**
Specify library name and version:  **assimp/1.0**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/assimp/5.x/conanfile.py [hunk 1]
```diff
@@ -145,6 +145,7 @@ def requirements(self):
 
         self.requires("minizip/1.2.11")
         self.requires("utfcpp/3.1.2")
+        self.requires("pugixml/1.11")
```
Comment (SpaceIm) [inline recipes/assimp/5.x/conanfile.py:148]: It's not a dependency of assimp < 5.1.0

Comment (sh0) [reply recipes/assimp/5.x/conanfile.py]: Thanks for spotting that, fixed. Didn't see your comment edit at first so ended up with two commits.

**Review by madebr — APPROVED** [review]

**Comment by SpaceIm** [comment]
Could you also ensure that vendored pugixml is not used by removing its folder?

```python
        # Take care to not use these vendored libs
        vendors = ["poly2tri", "rapidjson", "unzip", "utf8cpp", "zip"]
        if tools.Version(self.version) < "5.1.0":
            vendors.append("irrXML")
        else:
            vendors.append("pugixml")
        for vendor in vendors:
            tools.rmdir(os.path.join(self._source_subfolder, "contrib", vendor))
```

And by the way, draco is an optional dependency of gltf in assimp since 5.1.0 (disabled by default).

**Review by SSE4 — DISMISSED** [review]

**Comment by sh0** [comment]
@SpaceIm I finally had time to work on this PR. I removed the vendored `pugixml` (and `draco`) library in contrib.

Also added support for draco library. Right now draco dependency is pulled in whenever gltf is enabled. Theoretically we can make it optional by adding a new `with_draco` flag. Would that be needed as well?

**Review by SpaceIm — APPROVED** [review]
> Theoretically we can make it optional by adding a new with_draco flag. Would that be needed as well?

It could be an improvement yes.

**Review by danimtb — APPROVED** [review]

**Review by madebr — APPROVED** [review]

File: recipes/assimp/5.x/conanfile.py [hunk 2]
```diff
@@ -128,6 +128,12 @@ def _depends_on_poly2tri(self):
     def _depends_on_rapidjson(self):
         return self.options.with_gltf or self.options.with_gltf_exporter
 
+    @property
+    def _depends_on_draco(self):
+        if tools.Version(self.version) < tools.Version("5.1.0"):
```
Comment (prince-chrismc) [inline recipes/assimp/5.x/conanfile.py:133]: Is the double version required? I've never see that before 🤔

**Review by prince-chrismc — APPROVED** [review]

File: recipes/assimp/5.x/conanfile.py (hunk 2 above)
Comment (sh0) [reply recipes/assimp/5.x/conanfile.py:133]: Ah, yes, it is not needed indeed, but should work fine. I probably copied it from somewhere without thinking, my bad.

**Pull request merged**